### PR TITLE
Disable posts

### PR DESCRIPTION
**Pull request by roschaefer**
This is a horribly complicated implementation to allow only moderators to disable posts (only posts).

Reasons why this is so bad:
* No convenient access to database during tests (everything has to go through the API)
* I cannot change the generated mutation
* No filtering for `disabled: true` if disabled is a derived property

### CODE REVIEW

File: src/middleware/permissionsMiddleware.js [hunk 1]
```diff
@@ -41,6 +41,12 @@ const isAuthor = rule({ cache: 'no_cache' })(async (parent, args, { user, driver
   return authorId === user.id
 })
 
+const fromUserMatchesCurrentUser = rule({ cache: 'no_cache' })(async (parent, args, { user, driver }) => {
+  if (!user) return false
+  const { from: { id: fromId } } = args
+  return user.id === fromId
```
Comment (roschaefer) [inline src/middleware/permissionsMiddleware.js]: @appinteractive better than redundantly checking the `from` user is to override it with the current authenticated user. Too bad, we have to change the augmented `AddPostDisabledBy` mutation and then the generated resolver doesn't work anymore.

Comment (appinteractive) [reply src/middleware/permissionsMiddleware.js]: I would write our own resolver/mutation here anyway like I did with the (in)follow

Comment (appinteractive) [reply src/middleware/permissionsMiddleware.js]: Why not just adding something like that? `disable(Id: postId, type: Post)`

**Comment by mattwr18** [comment]
@roschaefer it looks good! was just having a look at the tests and reading the comments here... it looks like it was more work than we were anticipating?

**Comment by appinteractive** [comment]
🥳

**Pull request merged**